code/modules/badges/badges.dm

```diff
@@ -0,0 +1,88 @@
+/*
+
+Loads all badges from the badge_ranks.txt file.
+Format should be
+rank|icon
+Where:
+Rank - The name of the admin rank
+Icon - The name of the icon state in the badges.dmi file
+
+Special ranks:
+- Donator: Matches clients in the donator list
+- Mentor: Matches clients in the mentor list
+
+*/
+
+//Global list of badges
+GLOBAL_LIST_EMPTY(badge_data)
+
+/client
+	var/list/cached_badges = null
+
+//Loads the badge ranks
+/proc/load_badge_ranks()
+	//No badges
+	if(!CONFIG_GET(flag/badges))
+		return
+	//Load and parse data
+	GLOB.badge_data = json_decode(rustg_file_read("[global.config.directory]/badges.json"))
+	//Associate badges with admin ranks
+	for(var/datum/admin_rank/rank in GLOB.admin_ranks)
+		rank.badge_icon = GLOB.badge_data[rank.name]
+	//Yay
+	log_game("[LAZYLEN(GLOB.badge_data)] badges loaded successfully.")
+	//Reset everyones badges so they get reloaded.
+	for(var/client/C in GLOB.clients)
+		C.reset_badges()
+
+//Gets the badges attached to a client.
+/client/proc/get_badges()
+	//No badges
+	if(!CONFIG_GET(flag/badges))
+		return
+	//Send cached badges
+	if(islist(cached_badges))
```

Comment:
```suggestion
	if(!LAZYLEN(cached_badges))
```

Reply:
I want to check if its a list and not null, not if it has items in it

Reply:
Because isnull doesn't check if it's a list.